### PR TITLE
fix: MessageBar with no actions should have correct spacing

### LINKED ISSUE #30461: [Bug]: MessageBar bottom padding missing
### Library

React Northstar / v0 (@fluentui/react-northstar)

### System Info

```shell
no_need
```

### Are you reporting Accessibility issue?

None

### Reproduction

Make a two line messageBar

## Actual Behavior
image
## Expected Behavior
MessageBarBody text should have some padding on the bottom.

### Logs

_No response_

### Requested priority

Blocking

### Products/sites affected

_No response_

### Are you willing to submit a PR to fix?

yes

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

### PR DESCRIPTION
**Pull request by ling1726**
Design guidance has always considered MessageBar to include at least one action for accessibility requirements. However, there are still cases in product where MessageBar is used without any actions.

When no actions are rendered, the default secondary actions slot is no longer rendered (that provides correct spacing). In order to handle this edge case the MessageBar will render a new slot `bottomReflowSpacer` that sets the correct spacing in in the same grid area.

There should be no conflict with current secondary actions since the grid layout will render both in the same area and the spacer is just an empty div

Fixes #30461

## Related Issue(s)

- Fixes #

### CODE REVIEW

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)

  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30481/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">70</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30481/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">78</a></td>
            <td>10</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30481/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.regression.txt">Possible regression</a></td>
           </tr></table><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30481/perf-test-react-components/Avatar-mount_base.html">638</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30481/perf-test-react-components/Avatar-mount.html">592</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30481/perf-test-react-components/Button-mount_base.html">305</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30481/perf-test-react-components/Button-mount.html">296</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30481/perf-test-react-components/Field-mount_base.html">1121</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30481/perf-test-react-components/Field-mount.html">1127</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30481/perf-test-react-components/FluentProvider-mount_base.html">698</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30481/perf-test-react-components/FluentProvider-mount.html">723</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30481/perf-test-react-components/FluentProviderWithTheme-mount_base.html">72</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30481/perf-test-react-components/FluentProviderWithTheme-mount.html">77</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30481/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">54</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30481/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">64</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30481/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">70</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30481/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">78</a></td>
            <td>10</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30481/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.regression.txt">Possible regression</a></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30481/perf-test-react-components/MakeStyles-mount_base.html">855</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30481/perf-test-react-components/MakeStyles-mount.html">846</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30481/perf-test-react-components/Persona-mount_base.html">1773</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30481/perf-test-react-components/Persona-mount.html">1730</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30481/perf-test-react-components/SpinButton-mount_base.html">1414</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30481/perf-test-react-components/SpinButton-mount.html">1386</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports                                                                                                     |    Baseline (minified/GZIP) |                          PR |                                                                                                                                                                                                  Change |
| :-------------------------------------------------------------------------------------------------------------------- | --------------------------: | --------------------------: | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------: |
| <samp>react-message-bar</samp> <br /> <abbr title='bundle-size/Default.fixture.js'>MessageBar (all components)</abbr> | `25.181 kB`<br />`9.039 kB` | `25.566 kB`<br />`9.131 kB` | `385 B` <br />`92 B`  |

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                             |         Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ---------------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ButtonProviderAndTheme.fixture.js'>react-components: Button, FluentProvider & webLightTheme</abbr>              | `70.688 kB`<br />`20.464 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `217.02 kB`<br />`61.894 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>                            | `43.173 kB`<br />`14.298 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr>                                                         |   `7.902 kB`<br />`2.586 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/7b89afa6ca231f610fcb26f5d816e0b86faec3d9'>7b89afa6ca231f610fcb26f5d816e0b86faec3d9</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv9 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

File: apps/vr-tests-react-components/src/stories/MessageBar/MessageBar.stories.tsx [hunk 1]
```diff
@@ -64,6 +64,27 @@ export const Multiline = () => (
   </div>
```
Comment (fabricteam) [inline apps/vr-tests-react-components/src/stories/MessageBar/MessageBar.stories.tsx:64]: ## [🕵 fluentuiv9 Open the Visual Regressions report to inspect the affected screenshots]( https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-approval/index.html?organization=uifabric&project=fabricpublic&prId=30481&commitId=572aaf723d6321e59014c3388cdf419eaf5497a0&env=prod&clientType=FLUENTUI)


  <details><summary><b>MessageBar</b> 1 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|MessageBar.Multiline No Actions.default.chromium.png|0| Added|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

**Review by bsunderhus — APPROVED** [review]

**Pull request merged**